metaserver/src/main/scala/scala/meta/languageserver/Compiler.scala

```diff
@@ -40,7 +43,7 @@ class Compiler(
       .flatMap { config =>
         Observable.fromTask(
           Task(loadNewCompilerGlobals(config))
-            .zip(Task(indexDependencyClasspath(config)))
+            .zip(Task(indexDependencyClasspath(config.sourceJars)))
```

Comment:
I'm guessing this is safe for now but I worry this pattern will become dangerous in the future if the first task is mutating compiler globals and the second independent task is also working inside compiler.  I fear concurrency and mutation together in general though.

Reply:
The indexDependencyClasspath should be moved out of Compiler.scala since it doesn't do anything related to the compiler. Opened https://github.com/scalameta/language-server/issues/48